fix: Remove default pressure level variable_groups from training/*.yaml

timothyas opened the pull request:
## Description
Closes #435 

By removing any default configuration format for variable_groups within the training yamls.

## What problem does this change solve?

Very minor bugfix, in that it allows the user to easily define variable groups using whatever notation they like, e.g. they can do

```yaml
training:
  variable_groups:
    pl: [u, v, w]
```

or

```yaml
training:
  variable_groups:
    pl:
      param: [u, v, w]
```

without any OmegaConf errors.

## What issue or task does this change relate to?
#435 

##  Additional notes ##

Users relying on the defaults will have to update their configurations accordingly.


***As a contributor to the Anemoi framework, please ensure that your changes include unit tests, updates to any affected dependencies and documentation, and have been tested in a parallel setting  (i.e., with multiple GPUs). As a reviewer, you are also responsible for verifying these aspects and requesting changes if they are not adequately addressed. For guidelines about those please refer to https://anemoi.readthedocs.io/en/latest/***

By opening this pull request, I affirm that all authors agree to the [Contributor License Agreement.](https://github.com/ecmwf/codex/blob/main/Legal/contributor_license_agreement.md)


<!-- readthedocs-preview anemoi-training start -->
----
📚 Documentation preview 📚: https://anemoi-training--438.org.readthedocs.build/en/438/

<!-- readthedocs-preview anemoi-training end -->

<!-- readthedocs-preview anemoi-graphs start -->
----
📚 Documentation preview 📚: https://anemoi-graphs--438.org.readthedocs.build/en/438/

<!-- readthedocs-preview anemoi-graphs end -->

<!-- readthedocs-preview anemoi-models start -->
----
📚 Documentation preview 📚: https://anemoi-models--438.org.readthedocs.build/en/438/

<!-- readthedocs-preview anemoi-models end -->

mc4117 commented:
Good idea! I agree this would make the definition of variable groups much simpler. However, I was just wondering, if you remove default groups does the code still run or do you get an error because the default is undefined?

mc4117 commented on training/src/anemoi/training/config/training/default.yaml:
```diff
@@ -111,8 +111,6 @@ validation_metrics:
 
 variable_groups:
   default: sfc
-  pl:
```
Instead of removing these lines entirely, I would maybe keep them as a comment with some text, just so that there is an example of what the user can run

HCookie commented:
@timothyas Such is the joys of Omegaconf.
Commenting out the pressure level scalar seems fine, as I do agree that no default is better than a bad default. Feel free to make that modifcation to this PR. I would also suggest you add some docs / warnings to make it clear what is happening. This will also mean this PR becomes a breaking change.

timothyas commented:
That makes sense, thanks for the note @HCookie. I went ahead and took this path to what we discussed earlier: I removed the pressure scaler by default and added documentation to the scalers about what would need to be added to incorporate pressure scaling. Hopefully this helps evaluate this option, and I totally understand if you want to go the other route and change the format rather than break defaults.
